src/components/LogIn/UserPassForm/UserPassForm.tsx

```diff
@@ -0,0 +1,61 @@
+import { unwrapResult } from "@reduxjs/toolkit";
+import type { FormEvent } from "react";
+import { useEffect, useRef } from "react";
+
+import bakery from "juju/bakery";
+import { thunks as appThunks } from "store/app";
+import { actions as generalActions } from "store/general";
+import { getWSControllerURL } from "store/general/selectors";
+import { useAppDispatch, useAppSelector } from "store/store";
+
+import { Label } from "../types";
+
+interface LoginElements extends HTMLFormControlsCollection {
+  username: HTMLInputElement;
+  password: HTMLInputElement;
+}
+
+const UserPassForm = () => {
+  const dispatch = useAppDispatch();
+  const focus = useRef<HTMLInputElement>(null);
+  const wsControllerURL = useAppSelector(getWSControllerURL);
+
+  function handleSubmit(
+    e: FormEvent<HTMLFormElement & { elements: LoginElements }>,
+  ) {
+    e.preventDefault();
+    const elements = e.currentTarget.elements;
+    const user = elements.username.value;
+    const password = elements.password.value;
+    dispatch(generalActions.cleanupLoginErrors());
+    dispatch(
+      generalActions.storeUserPass({
+        wsControllerURL,
+        credential: { user, password },
+      }),
+    );
+    if (bakery) {
+      dispatch(appThunks.connectAndStartPolling())
+        .then(unwrapResult)
+        .catch((error) => console.error(Label.POLLING_ERROR, error));
+    }
+  }
+
+  useEffect(() => {
+    focus.current?.focus();
+  }, []);
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <label htmlFor="username">Username</label>
+      <input type="text" name="username" id="username" ref={focus} />
```

Comment:
There's currently a bug where the username doesn't always get focus, so as a tiny fix you could replace `ref={focus}` with `autoFocus` and remove the useRef and useEffect above.